cmd/state-custom-installer/runInstall.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	rt "runtime"
+
+	"github.com/ActiveState/cli/internal/analytics/client/blackhole"
+	"github.com/ActiveState/cli/internal/assets"
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/runbits/buildlogfile"
+	"github.com/ActiveState/cli/internal/subshell"
+	"github.com/ActiveState/cli/internal/subshell/sscommon"
+	"github.com/ActiveState/cli/internal/testhelpers/outputhelper"
+	"github.com/ActiveState/cli/internal/unarchiver"
+	"github.com/ActiveState/cli/pkg/platform/runtime"
+	"github.com/ActiveState/cli/pkg/platform/runtime/setup/events"
+	"github.com/ActiveState/cli/pkg/platform/runtime/target"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+func runInstall(out output.Outputer, params *Params) error {
+
+	tempDir, err := ioutil.TempDir("", "artifacts-")
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// #FIX: Need to uncomment this for production
+	defer os.RemoveAll(tempDir)
+
+	fmt.Printf("Temp directory is: %s\n", tempDir)
+
+	installToDir := params.path
+	artifactsTarGz := "artifacts.tar.gz"
+	assetsPath := filepath.Join(tempDir, "assets")
+	artifactsPath := filepath.Join(tempDir, "artifacts")
+	licenseFilePath := filepath.Join(assetsPath, "LICENSE.txt")
+
+	// Double check if installation directory exists already
+	if fileutils.DirExists(installToDir) {
+		empty, err := fileutils.IsEmptyDir(installToDir)
+		if err == nil && !empty {
+			configureEnvironmentAccepted, err := getConfirmation("Installation directory is not empty, install anyway?")
+			if err != nil {
+				return err
+			}
+
+			if !configureEnvironmentAccepted {
+				return errs.New("Installation directory not empty, installation aborted")
+			}
+		}
+	}
+
+	if err := os.Mkdir(assetsPath, os.ModePerm); err != nil {
+		return err
+	}
+
+	if err := os.Mkdir(artifactsPath, os.ModePerm); err != nil {
+		return err
+	}
+
+	err = unzip(os.Args[0], assetsPath)
```

Comment:
Use `params` instead. Once you're in the command execution you generally should not be using `os.Args` anymore.